FeedStoreChallenge/CoreData/ManagedCache.swift

```diff
@@ -0,0 +1,28 @@
+//
+//  ManagedCache+CoreDataClass.swift
+//  FeedStoreChallenge
+//
+//  Created by Stas Lee on 11/8/21.
+//  Copyright © 2021 Essential Developer. All rights reserved.
+//
+//
+
+import Foundation
+import CoreData
+
+@objc(ManagedCache)
+class ManagedCache: NSManagedObject {
```

Comment:
`ManagedCache` is not supposed to be subclassed, so it should be marked as `final`:
```suggestion
final class ManagedCache: NSManagedObject {
```

Reply:
Got a question here. Not supported to subclasses in terms of CoreData models as dto ? Thus we are restricting the subclassing functionality here @thiagopenna

Reply:
Hey @gitlees, I mean that based on our current design, there's no immediate need to subclass `ManagedCache`. Knowing that, we can increase performance by reducing dynamic dispatch, using the `final` keyword. From Apple's Swift [Blog](https://developer.apple.com/swift/blog/?id=27):

> [...] dynamic dispatch, increases language expressivity at the cost of a constant amount of runtime overhead for each indirect usage. In performance sensitive code such overhead is often undesirable. [...]
>
> The final keyword is a restriction on a class, method, or property that indicates that the declaration cannot be overridden. This allows the compiler to safely elide dynamic dispatch indirection. [...]

If we need to subclass in the future (for a good reason), we can remove this optimization.